mlir/lib/Analysis/Presburger/IntegerRelation.cpp

```diff
@@ -1600,21 +1598,14 @@ std::optional<DynamicAPInt> IntegerRelation::getConstantBoundOnDimSize(
   // of the symbolic variables (+ constant).
   int eqPos = findEqualityToConstant(*this, pos, /*symbolic=*/true);
   if (eqPos != -1) {
-    auto eq = getEquality(eqPos);
-    // If the equality involves a local var, punt for now.
-    // TODO: this can be handled in the future by using the explicit
-    // representation of the local vars.
-    if (!std::all_of(eq.begin() + getNumDimAndSymbolVars(), eq.end() - 1,
-                     [](const DynamicAPInt &coeff) { return coeff == 0; }))
-      return std::nullopt;
-
     // This variable can only take a single value.
     if (lb) {
       // Set lb to that symbolic value.
-      lb->resize(getNumSymbolVars() + 1);
+      lb->resize(getNumSymbolVars() + getNumLocalVars() + 1);
```

Comment:
This is actually not right. The local variable itself may not be symbolic, and as a result, you can't conclude that the `pos^th` variable can only take a single value contrary to the comment above. You need to find out what the local variable is.

Eg:
```
d1 - s0 - l0 = 1,
d0 - l0 = 0
0 <= d0 <= s1
0 <= d1 < s2
```
`getConstantBoundOnDimSize` on `d1` for the above set should return nothing, it's not bounded by a constant size, but you are returning a bounded size of one post this change.